scripts/kind-e2e/e2e.sh

```diff
@@ -322,16 +331,108 @@ setup_custom_cni
 if [[ $3 = true ]]; then
     enable_logging
 fi
-install_helm
-if [[ $4 = true ]]; then
-    enable_kubefed
+
+if [[ $5 = operator ]]; then
+    operator=true
+    # TODO: Convert these Broker-deploy steps to use Operator instead of Helm
+    install_helm
+    kind_import_images
+    # NB: This fn will skip/exit if clusters CRD already exists
+    setup_broker
+    # Verify SubM Broker secrets
+    collect_subm_vars cluster1
+    verify_subm_broker_secrets cluster1
+
+    for i in 2 3; do
+      # Create CRDs required as prerequisite submariner-engine
+      # TODO: Eventually OLM should handle this
+      create_subm_endpoints_crd cluster$i
+      verify_endpoints_crd cluster$i
+      create_subm_clusters_crd cluster$i
+      verify_clusters_crd cluster$i
+      if [[ $also_routeagent = true ]]; then
+        create_routeagents_crd cluster$i
+        verify_routeagents_crd cluster$i
+      fi
+
+      # Add SubM gateway labels
+      add_subm_gateway_label cluster$i
+      # Verify SubM gateway labels
+      verify_subm_gateway_label cluster$i
+
+      # Deploy SubM Operator
+      deploy_subm_operator cluster$i
+      # Verify SubM CRD
+      verify_subm_crd cluster$i
+      # Verify SubM Operator
+      verify_subm_operator cluster$i
+      # Verify SubM Operator pod
+      verify_subm_op_pod cluster$i
+      # Verify SubM Operator container
+      verify_subm_operator_container cluster$i
+
+      # Collect SubM vars for use in SubM CRs
+      collect_subm_vars cluster$i
+      if [[ $also_engine = true ]]; then
+        # FIXME: Rename all of these submariner-engine or engine, vs submariner
+        # Create SubM CR
+        create_subm_cr cluster$i
+        # Deploy SubM CR
+        deploy_subm_cr cluster$i
+        # Verify SubM CR
+        verify_subm_cr cluster$i
+        # Verify SubM Engine Pod
+        verify_subm_engine_pod cluster$i
+        # Verify SubM Engine container
+        verify_subm_engine_container cluster$i
+        # Verify Engine secrets
+        verify_subm_engine_secrets cluster$i
+      fi
+      if [[ $also_routeagent = true ]]; then
+        # Create Routeagent CR
+        create_routeagent_cr cluster$i
+        # Deploy Routeagent CR
+        deploy_routeagent_cr cluster$i
+        # Verify Routeagent CR
+        verify_routeagent_cr cluster$i
+        # Verify SubM Routeagent Pods
+        verify_subm_routeagent_pod cluster$i
+        # Verify SubM Routeagent container
+        verify_subm_routeagent_container cluster$i
+        # Verify Routeagent secrets
+        verify_subm_routeagent_secrets cluster$i
+      fi
+    done
+
+    deploy_netshoot_cluster2
+    deploy_nginx_cluster3
+
+    test_connection
```

Comment:
These two should be after the condition block since you execute them in both cases

Reply:
Good suggestion, thanks, done in the latest version. :+1: